packages/core/graph/src/AdjacencyList.js

```diff
@@ -0,0 +1,1195 @@
+    if (prevEdge && nextEdge) this.linkEdge(prevEdge, nextEdge, NEXT_HASH);
+    else if (prevEdge) this.unlinkEdge(prevEdge, NEXT_HASH);
+    else if (nextEdge) this.linkEdge(hash, nextEdge);
+    else this.unlinkEdge(hash);
+
+    // Splice the removed edge out of the linked list of incoming edges.
+    if (previousIn && nextIn) {
+      this.linkEdge(previousIn, nextIn, NEXT_IN);
+      this.linkEdge(nextIn, previousIn, PREV_IN);
+    } else if (previousIn) {
+      this.unlinkEdge(previousIn, NEXT_IN);
+    } else if (nextIn) {
+      this.unlinkEdge(nextIn, PREV_IN);
+    }
+
+    // Splice the removed edge out of the linked list of outgoing edges.
+    if (previousOut && nextOut) {
+      this.linkEdge(previousOut, nextOut, NEXT_OUT);
+      this.linkEdge(nextOut, previousOut, PREV_OUT);
+    } else if (previousOut) {
+      this.unlinkEdge(previousOut, NEXT_OUT);
+    } else if (nextOut) {
+      this.unlinkEdge(nextOut, PREV_OUT);
+    }
+
+    // Update the terminating node's first and last incoming edges.
+    if (firstIn === edge) this.setEdge(to, nextIn, FIRST_IN);
+    if (lastIn === edge) this.setEdge(to, previousIn, LAST_IN);
+
+    // Update the originating node's first and last outgoing edges.
+    if (firstOut === edge) this.setEdge(from, nextOut, FIRST_OUT);
+    if (lastOut === edge) this.setEdge(from, previousOut, LAST_OUT);
+
+    this.#edges[edge + TYPE] = 0;
+    this.#edges[edge + FROM] = 0;
+    this.#edges[edge + TO] = 0;
+    this.#edges[edge + NEXT_HASH] = 0;
+    this.#edges[edge + NEXT_IN] = 0;
+    this.#edges[edge + PREV_IN] = 0;
+    this.#edges[edge + NEXT_OUT] = 0;
+    this.#edges[edge + PREV_OUT] = 0;
+
+    this.#edges[COUNT]--;
+    this.#edges[DELETES]++;
+
+    this.#typeMap?.delete(from, to, type);
+  }
+
+  hasInboundEdges(to: NodeId): boolean {
+    return Boolean(this.getEdge(to, FIRST_IN));
+  }
+
+  getInboundEdgesByType(to: NodeId): {|type: TEdgeType, from: NodeId|}[] {
+    let edges = [];
+    let typeMap = this.#typeMap || (this.#typeMap = new TypeMap(this));
+    for (let [type, nodes] of typeMap.getConnectedTo(to)) {
+      for (let from of nodes) {
+        edges.push({type: (type: any), from});
+      }
+    }
+    return edges;
+  }
+
+  getOutboundEdgesByType(from: NodeId): {|type: TEdgeType, to: NodeId|}[] {
+    let edges = [];
+    let typeMap = this.#typeMap || (this.#typeMap = new TypeMap(this));
+    for (let [type, nodes] of typeMap.getConnectedFrom(from)) {
+      for (let to of nodes) {
+        edges.push({type: (type: any), to});
+      }
+    }
+    return edges;
+  }
+
+  /**
+   *
+   */
+  getEdges(
+    from: NodeId,
+    type:
+      | AllEdgeTypes
+      | TEdgeType
+      | NullEdgeType
+      | Array<TEdgeType | NullEdgeType> = 1,
+  ): $ReadOnlySet<NodeId> {
+    return new Set(this.getNodesConnectedFrom(from, type));
+  }
+
+  /**
+   * Get the list of nodes connected from this node.
+   */
+  getNodesConnectedFrom(
```

Comment:
These functions can probably be renamed to `getNodeIdsConnected..` to match the ones in `Graph.js`.